scapy/arch/bpf/supersocket.py

```diff
@@ -241,6 +241,8 @@ def get_frame(self):
         """Get a frame or packet from the received list"""
         if self.received_frames:
             return self.received_frames.pop(0)
+        else:
```

Comment:
Avoid a race condition that could lead this function to return None.

Reply:
That's nothing but you don't need `else` after `return`.